graphql-server/src/schemas/queries/patients.ts

```diff
@@ -177,3 +190,88 @@ export async function queryDashboardPatients({
     await session.close();
   }
 }
+
+export async function queryPatientIdsTriplets(searchVals: Array<string>) {
+  const searchValsWithoutCDash = searchVals.map((searchVal) =>
+    searchVal.startsWith("C-") ? searchVal.slice(2) : searchVal
+  );
+  const searchValList = searchValsWithoutCDash
+    .map((searchVal) => `'${searchVal}'`)
+    .join(",");
+  const query = `
+    SELECT
+      CMO_PATIENT_ID,
+      DMP_PATIENT_ID,
+      MRN
+    FROM
+      ${PHI_ID_MAPPING_TABLE}
+    WHERE
+      DMP_PATIENT_ID IN (${searchValList})
+      OR MRN IN (${searchValList})
+      OR CMO_PATIENT_ID IN (${searchValList})
+  `;
+  const patientIdsTriplets = await queryDatabricks<PatientIdsTriplet>(query);
+  patientIdsTriplets.forEach((patientIdTriplet) => {
+    patientIdTriplet.CMO_PATIENT_ID = `C-${patientIdTriplet.CMO_PATIENT_ID}`;
+  });
+  return patientIdsTriplets;
+}
```

Comment:
Note that the C- is removed from CMO Patient ID before querying from Databricks then gets added back afterwards.